processing/src/main/java/org/apache/druid/query/scan/ScanQueryOrderBySequence.java

```diff
@@ -0,0 +1,182 @@
+{
+  public ScanQueryOrderBySequence(
+      QueryPlus<ScanResultValue> queryPlus,
+      QueryProcessingPool queryProcessingPool,
+      Iterable<QueryRunner<ScanResultValue>> queryables,
+      ResponseContext responseContext
+  )
+  {
+    super(new ScanQueryOrderByIteratorMaker(queryPlus, queryProcessingPool, queryables, responseContext));
+  }
+
+  static class ScanQueryOrderByIteratorMaker
+      implements BaseSequence.IteratorMaker<ScanResultValue, Iterator<ScanResultValue>>
+  {
+    private static final Logger log = new Logger(ScanQueryOrderByIteratorMaker.class);
+    private final QueryPlus<ScanResultValue> threadSafeQueryPlus;
+    private final ResponseContext responseContext;
+    private final Iterable<QueryRunner<ScanResultValue>> queryables;
+    private final QueryProcessingPool queryProcessingPool;
+
+    ScanQueryOrderByIteratorMaker(
+        QueryPlus<ScanResultValue> queryPlus,
+        QueryProcessingPool queryProcessingPool,
+        Iterable<QueryRunner<ScanResultValue>> queryables,
+        ResponseContext responseContext
+    )
+    {
+      this.threadSafeQueryPlus = queryPlus.withoutThreadUnsafeState();
+      this.responseContext = responseContext;
+      this.queryables = queryables;
+      this.queryProcessingPool = queryProcessingPool;
+    }
+
+    @Override
+    public Iterator<ScanResultValue> make()
+    {
+      ScanQuery query = (ScanQuery) threadSafeQueryPlus.getQuery();
+      QueryContext queryContext = QueryContext.of(query.getContext());
+      List<ListenableFuture<ScanResultValue>> futures =
+          Lists.newArrayList(
+              Iterables.transform(
+                  queryables,
+                  input -> {
+                    if (input == null) {
+                      throw new ISE("Null queryRunner! Looks to be some segment unmapping action happening");
+                    }
+
+                    return queryProcessingPool.submitRunnerTask(
+                        new AbstractPrioritizedQueryRunnerCallable<ScanResultValue, ScanResultValue>(
+                            queryContext.getPriority(),
+                            input
+                        )
+                        {
+                          @Override
+                          public ScanResultValue call()
+                          {
+                            try {
+                              Sequence<ScanResultValue> result = input.run(threadSafeQueryPlus, responseContext);
+                              if (result == null) {
+                                throw new ISE("Got a null result! Segments are missing!");
+                              }
+
+                              Iterator<ScanResultValue> it = result.toList().iterator();
+                              List<List<Object>> eventList = new ArrayList<>();
+                              List<String> columns = new ArrayList<>();
+                              String segmentId = null;
+                              while (it.hasNext()) {
```

Comment:
Maybe a comment to explain what we're doing? Looks like we read a set of scan result values and combine their rows into one bug scan result value. Again, the idea with scan query is to limit batch size. Does something else where break the results back into batches? Or, maybe copies these values into a sorter? If copy into a sorter, should that be done here, then later merge the results?